src/lib.rs

```diff
@@ -1665,10 +1671,33 @@ impl GooseAttack {
                 }
 
                 // Flush request statistics collected prior to all user threads running
-                if self.configuration.reset_stats && !statistics_reset {
-                    info!("statistics reset...");
-                    self.stats.requests = HashMap::new();
-                    statistics_reset = true;
+                if !users_launched {
+                    users_launched = true;
+                    if !self.configuration.no_reset_stats {
```

Comment:
Given how deep this is indented it belongs into a helper function -- the compiler will likely inline it, so performance is the same, but readability is better.

Reply:
this would be a major refactor of unrelated code, is this the right time to do that?  (we're inside a class, inside a thread, inside a loop)

Reply:
This is all newly added code - that all only deals with `self.`

I am not seeing why it would be a refactor?

Reply:
it's just a smaller part of a bigger needed cleanup -- ie, the thread can be in its own function, the loop can be in its own function, seems good to clean it all up at the same time?

Reply:
https://github.com/tag1consulting/goose/issues/136